vit_keras/utils.py

```diff
@@ -60,39 +60,44 @@ def read(filepath_or_buffer: ImageInputType, size, timeout=None):
     return cv2.resize(image, (size, size))
 
 
-def apply_embedding_weights(target_layer, source_weights):
+def apply_embedding_weights(target_layer, source_weights, num_x_patches, num_y_patches):
     """Apply embedding weights to a target layer.
 
     Args:
         target_layer: The target layer to which weights will
             be applied.
         source_weights: The source weights, which will be
             resized as necessary.
+        num_x_patches: Number of patches in height of image.
+        num_y_patches: Number of patches in width of image.
     """
     expected_shape = target_layer.weights[0].shape
     if expected_shape != source_weights.shape:
         token, grid = source_weights[0, :1], source_weights[0, 1:]
         sin = int(np.sqrt(grid.shape[0]))
```

Comment:
As the grid is already flattened, the information about the number of pixels is only in the 0th dimension of the grid variable, and the 1st dimension is the embedding space. Hence, using `sin_y, sin_x = map(int, np.sqrt(grid.shape[:2]))` will return the number of patches in `sin_y`, however, it will return sqrt of embedding dimension in `sin_x`, which is independent of number of patches.  One way can be `sin_y = sin_x =  int(np.sqrt(grid.shape[0]))`. But it still does not provide the support for rectangular input patches. Hence, doing any such thing right now will only create an extra variable without any usefulness.

Reply:
Ah, yes, that makes sense thank you!